compiler.rb

```diff
@@ -155,8 +154,15 @@
     next
   end
 
-  product_api, provider_config, = \
-    Provider::Config.parse(provider_yaml_path, product_api, version)
+  if File.exist?(provider_yaml_path)
+    product_api, provider_config, = \
+      Provider::Config.parse(provider_yaml_path, product_api, version)
+  end
+  # Load any dynamic overrides passed in with -r
+  if File.exist?(provider_override_path)
+    product_api, provider_config, = \
+      Provider::Config.parse(provider_override_path, product_api, version)
```

Comment:
What do you think about moving validation out of this step? Ideally, I would think this would be the proper flow:

1) Grab the YAML
2) Apply the regular overrides
3) Apply the dynamic overrides
4) Validate + add default values.

Reply:
Right now, the flow is:

1) Grab the YAML
2) Apply the regular overrides
3) Validate + add default values.
4) Apply the dynamic overrides
5) Validate + add default values (defaults shouldn't change, so this is just validating the dynamic overrides)

Reply:
Yup that makes sense. I'll play around with pulling it out but I suspect it might be more difficult than it sounds. 

Unrelated: Is it a standard ruby thing to have a `validate` method which mutates a classes members?

Reply:
follow up: Even with a validation refactor it still doesn't cover this scenario. It would require rethinking the concept of defaulting fields in the override.validte call which I don't think is worth doing in this change.